spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/ControlInfoValidation.scala

```diff
@@ -26,28 +28,43 @@ object ControlInfoValidation {
   private val rawFieldName = "raw"
   private val sourceFieldName = "source"
 
-  case class FieldCounts(rawCount: Long, sourceCount: Long)
-
   /**
    * Adds metadata about the number of records in raw and source data by checking Atum's checkpoints first.
    */
-  def addRawAndSourceRecordCountsToMetadata(): Unit = {
-    val fields = getFieldCounts(Atum.getControlMeasure.checkpoints)
-    Atum.setAdditionalInfo(s"${rawFieldName}_record_count" -> fields.rawCount.toString)
-    Atum.setAdditionalInfo(s"${sourceFieldName}_record_count" -> fields.sourceCount.toString)
-  }
-
-  def getFieldCounts(checkpoints: Seq[Checkpoint]): FieldCounts = {
+  def addRawAndSourceRecordCountsToMetadata(controlInfoValidation: String, log: Logger): Unit = {
+    val checkpoints = Atum.getControlMeasure.checkpoints
     val checkpointRawRecordCount = getCountFromGivenCheckpoint(rawFieldName, checkpoints)
     val checkpointSourceRecordCount = getCountFromGivenCheckpoint(sourceFieldName, checkpoints)
+
+    try {
+        validateFields(checkpointRawRecordCount, checkpointSourceRecordCount)
+    } catch {
+      case ex: ValidationException => controlInfoValidation match {
+        case "strict" => throw ex
+        case "warning" => log.warn(ex.msg)
+        case _ =>
+      }
+    } finally {
+      checkpointRawRecordCount
+        .foreach(e => Atum.setAdditionalInfo(s"${rawFieldName}_record_count" -> e.toString))
```

Comment:
Not sure. Why is this moved to additional info?

Reply:
These fields in additional info help Menas Monitoring and [possibly] SPOT to parse _INFO files. Quite convenient.